filesystem.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+)
+
+type fileSystem interface {
+	Open(name string) (file, error)
+	Stat(name string) (os.FileInfo, error)
+	Create(name string) (file, error)
+	ReadFile(string) ([]byte, error)
+}
+
+type file interface {
+	io.Closer
+	io.Reader
+	io.ReaderAt
+	io.Seeker
+	io.Writer
+
+	Stat() (os.FileInfo, error)
+	WriteString(string) (int, error)
+}
+
+type osFS struct{}
+
+func (osFS) Open(name string) (file, error)        { return os.Open(name) }
+func (osFS) Stat(name string) (os.FileInfo, error) { return os.Stat(name) }
+func (osFS) Create(name string) (file, error)      { return os.Create(name) }
+func (osFS) ReadFile(name string) ([]byte, error)  { return ioutil.ReadFile(name) }
```

Comment:
I also think it might be worthwhile to add a comment to the top of this file explaining it's purpose / why we are abstracting the file system methods.